internal/websocket/e2e_test.go

```diff
@@ -44,14 +48,19 @@ func TestEndToEnd(t *testing.T) {
 				require.NoError(err)
 
 				mt, got, err := c.Read(ctx)
+				// server will halt until the websocket closes resulting in a EOF
+				if finished && errors.Is(err, io.EOF) {
```

Comment:
No need to use `errors.Is` here.  In fact, you really shouldn't.  `io.Reader.Read` is documented as returning *exactly* `io.EOF` in this case.  I realize the linter will complain, which is why I hate our linter :)

https://pkg.go.dev/io#Reader